src/utils/toasts.js

```diff
@@ -17,15 +17,32 @@
     with this program; if not, write to the Free Software Foundation, Inc.,
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
+import { NavigationActions } from 'react-navigation';
 import Toast from 'components/Toast';
+import { WALLET_SETTINGS } from 'constants/navigationConstants';
+import { navigate } from 'services/navigation';
+import type { Account } from 'models/Account';
 
-export function toastWalletBackup(isWalletBackedUp: boolean) {
-  if (!isWalletBackedUp) {
-    Toast.show({
-      message: 'Go to settings on the home screen and complete the wallet backup. Pillar cannot help you retrieve your wallet if it is lost.', // eslint-disable-line max-len
-      type: 'warning',
-      title: 'Please ensure you backup your wallet now',
-      autoClose: false,
-    });
+const BACKUP_MESSAGE =
+  'Go to wallet settings on the assets screen and complete the wallet backup. ' +
+  'Pillar cannot help you retrieve your wallet if it is lost.';
+
+export const toastWalletBackup = (isWalletBackedUp: boolean, wallet: Account) => {
+  if (isWalletBackedUp) {
+    return;
   }
-}
+
+  Toast.show({
+    message: BACKUP_MESSAGE,
+    type: 'warning',
+    title: 'Please ensure you backup your wallet now',
+    autoClose: false,
+    onPress: () => {
+      const action = NavigationActions.navigate({
+        routeName: WALLET_SETTINGS,
+        params: { wallet },
```

Comment:
As we're trying to avoid wallet `privateKey` or `mnemonic` to not use where it's not needed I would highly suggest to remove those from `wallet` object or only pass here what's really needed in WalletSettings screen and it's only one param – `wallet.type`.

Reply:
Or even better to pass already truncated wallet object (with type only) to this toast constructor.

Reply:
Pls update your code according to changes in this PR: https://github.com/pillarwallet/pillarwallet/pull/1436

Reply:
Updated